### ISSUE TITLE
Add rustfmt.toml

### ISSUE THREAD

**Issue by rimutaka** [shuttle-hq/shuttle#98]
My VSCode kept re-formatting the code on Ctrl+Shift+I creating huge diffs. All contributors should share the same formatting rules.

See https://rust-lang.github.io/rustfmt/

The alternative is to accept that the diffs will contain formatting changes potentially reverting back and forth depending on who's got what version of `rustfmt` installed.

**Comment by chesedo** [comment]
We seem to have fixed this on #41 and then applied it to all the code in #107.

However I get a bunch of the following types of lines on `main`:
```
Warning: can't set `imports_layout = Vertical`, unstable features are only available in nightly channel.
```

@brokad will we close this issue after taking care of the warnings?

**Comment by chesedo** [comment]
An example of the warnings on our CI

https://github.com/getsynth/shuttle/runs/5814293026?check_suite_focus=true#step:4:4
